rfd/0202-db-multi-session-mfa.md

```diff
@@ -0,0 +1,309 @@
+---
+authors: STeve Huang (xin.huang@goteleport.com)
+state: draft
+---
+
+RFD 202 - Database Multi-session MFA
+
+Required Approvers:
+- Engineering: @r0mant && @codingllama
+
+## What
+
+Allows executing multiple database connections with a single MFA tap.
+
+## Why
+
+Teleport today supports per-session MFA for enhanced security. However, when a
+user needs to run queries on multiple database hosts simultaneously, they have
+to perform a tap for every connection.
+
+A relaxed mode of per-session MFA will be introduced so that an MFA challenge is
+still required for connecting to target databases but the MFA response can be
+reused for a short period of time without the need to prompt the user again in
+that period.
+
+In addition to expanding MFA functionalities, a new `tsh` command will be
+introduced to assist executing queries on multiple database servers using a
+single CLI command.
+
+## Details
+
+### UX
+
+I want to execute a database query on multiple database services without being
+prompted by MFA for every database service. The Teleport role that grants
+database access can be updated as below:
+```diff
+kind: role
+version: v7
+metadata:
+  name: example-role-with-mfa
+spec:
+  options:
+-   require_session_mfa: true
++   # "multi-session" allows reuse of the MFA response where applicable (e.g.
++   # a single WebAuthn prompt for `tsh db exec` on multiple databases), and
++   # fallbacks to per-session MFA in other places where multi-session is not
++   # supported.
++   require_session_mfa: "multi-session"
```

Comment:
Unless I missed it I don't think I saw the information about the MFA response cache policy and behavior:

- How long will it be cached for?
- Will the interval be configurable server-side (FWIW I think it doesn't need to be, at least until we need it)?
- Will it be cached per user/role/connection/dbs? I.e. if I run a `tsh db exec`, and then run another `tsh db exec` for the same or different set of databases right away, will I get prompted for MFA again or not?
- Can you be prompted for MFA again in the middle of `tsh db exec` if execution takes a long time?

Reply:
Follow-up: I see the mention below that it's 5 minutes. Nevertheless, I think a separate section specifically explaining caching behavior from UX perspective will be useful.

Reply:
the interval is currently hard-coded to 5 minutes on server-side. i've expanded that section a little bit. And i've added caching behavior to the UX examples.